rtos/Mutex.cpp

```diff
@@ -58,11 +59,30 @@ osStatus Mutex::lock(uint32_t millisec) {
 }
 
 bool Mutex::trylock() {
-    if (osMutexAcquire(_id, 0) == osOK) {
-        _count++;
+    return trylock_for(0);
+}
+
+bool Mutex::trylock_for(uint32_t millisec) {
+    osStatus status = lock(millisec);
+    if (status == osOK) {
```

Comment:
I suppose _count should be incremented in case of success.

Reply:
`lock(millisec)` already does that - this is just munging the return value (and asserting there's no programming error).